src/utils/sourcemaps.rs

```diff
@@ -578,19 +627,68 @@ impl SourceMapProcessor {
         }
     }
 
+    /// Old versions of Sentry cannot assemble artifact bundles straight away, they require
+    /// that those bundles are associated to a release.
+    ///
+    /// This function checks if given chunk upload options about the server's capabilities
+    /// and only creates a release if the server requires that.
+    fn initialize_legacy_release_upload(&self, context: &UploadContext) -> Result<()> {
+        // if the remote sentry service supports artifact bundles, we don't
+        // need to do anything here.  Artifact bundles will also only work
+        // if a project is provided which is technically unnecessary for the
+        // legacy upload though it will rarely succeed.
+        if context.project.is_some()
+            && context.chunk_upload_options.map_or(false, |x| {
+                x.supports(ChunkUploadCapability::ArtifactBundles)
+            })
+        {
+            return Ok(());
+        }
+
+        // TODO: make this into an error later down the road
+        if context.project.is_none() {
+            eprintln!(
+                "{}",
+                style(
+                    "warning: no project specified. \
+                    While this upload will succeed it will be unlikely that \
+                    this is what you wanted. Future versions of sentry will \
+                    require a project to be set."
+                )
+                .red()
+            );
+        }
+
+        if let Some(version) = context.release {
```

Comment:
I really dislike this `version`/`release` naming confusion

Reply:
you and me both :(